src/encoded/static/components/report.js

```diff
@@ -282,53 +283,190 @@ ColumnHeader.propTypes = {
+        // if the setState callback returned no properties setState becomes a null op, so the above
+        // test could be done inside the setState callback. The React docs don't say what happens
+        // if you return no properties (https://reactjs.org/docs/react-component.html#setstate) so
+        // I avoided doing this.
+        this.setState((prevState) => {
+            // Toggle the `visible` state corresponding to the column whose checkbox was toggled.
+            // Then set that as the new React state which causes a redraw of the modal with all the
+            // checkboxes in the correct state.
+            prevState.columns[columnPath].visible = !prevState.columns[columnPath].visible;
+            return { columns: prevState.columns };
+        });
     }
 
-    toggleColumn(path) {
-        this.props.toggleColumn(path);
+    submitHandler() {
+        // Called when the user clicks the Select button in the column checkbox modal, which
+        // sets a new state for all checked report columns.
+        this.props.setColumnState(this.state.columns);
+    }
+
+    handleSelectAll() {
+        // Called when the "Select all" button is clicked.
+        this.setState((prevState) => {
+            // For every column in this.state.columns, set its `visible` property to true.
+            Object.keys(prevState.columns).forEach((columnPath) => {
+                prevState.columns[columnPath].visible = true;
+            });
+            return { columns: prevState.columns };
+        });
+    }
+
+    handleSelectOne() {
+        // Called when the "Select (first) only" button is clicked.
+        this.setState((prevState) => {
+            // Set all columns to invisible first.
+            const columnPaths = Object.keys(prevState.columns);
+            columnPaths.forEach((columnPath) => {
+                prevState.columns[columnPath].visible = false;
+            });
+
+            // Now set the column for the first key to true so that only it's selected.
+            prevState.columns[columnPaths[0]].visible = true;
+            return { columns: prevState.columns };
+        });
+    }
+
+    // Called when the sorting option gets changed.
+    handleSortChange(sortOption) {
+        this.setState({ sortOption });
     }
 
     render() {
-        const columnPaths = Object.keys(this.props.columns);
+        const { columns } = this.state;
+        const firstColumnKey = Object.keys(columns)[0];
+
+        // Get the column paths, sorting them by the corresponding column title if the user asked
+        // for that.
+        let columnPaths = Object.keys(columns);
+        if (this.state.sortOption === 'alpha') {
+            columnPaths = columnPaths.sort((aKey, bKey) => {
+                const aTitle = columns[aKey].title.toLowerCase();
+                const bTitle = columns[bKey].title.toLowerCase();
+                return (aTitle < bTitle ? -1 : (bTitle < aTitle ? 1 : 0));
+            });
```

Comment:
It would be best to define this sort comparison function at module scope, to avoid constructing a closure every time render is called.